go/example.go

```diff
@@ -93,7 +96,9 @@ func main() {
 	client, err := flight.NewClientWithMiddleware(
 		net.JoinHostPort(config.Host, config.Port),
 		nil,
-		nil,
+		[]flight.ClientMiddleware{
```

Comment:
Why do we need in this PR? Why not it was not needed before?

Reply:
This middleware handles cookie-based sessions. When the server responds to the SetSessionOptions call with a Set-Cookie header with the SessionID, the middleware ensures that the client includes it in all subsequent calls. From what I understand, the server then looks up the Project ID based on this Session ID. Without it, the next call generates a new SessionID which makes the server revert to using the default project. It wasn't needed before because there were no cookies that needed to be handled I think.